File: js/examples/triad_example.js
#!/usr/bin/env node
/**
 * Runs Fire's bundled "triad" flow: two hot agents (one agreeable, one
 * contrarian) answer your question in parallel, you review both, then a
 * cool synthesizer writes the final answer. Mirrors
 * python/examples/triad_example.py and php/examples/triad_example.php
 * exactly.
 *
 * Usage:
 *     FIRE_TOKEN=fire_sk_... node examples/triad_example.js "Should cities ban cars downtown?"
 *
 * Get a token from whoever gave you access to this SDK, or self-serve one
 * with a tier code (see README.md's "Getting a token" section).
 */

import { createInterface } from 'node:readline/promises';
import { stdin, stdout } from 'node:process';

import { FireClient } from '../src/FireClient.js';

async function main() {
  const token = process.env.FIRE_TOKEN;
  if (!token) {
    console.error('Set FIRE_TOKEN in your environment first.');
    process.exit(1);
  }

  const question = process.argv.slice(2).join(' ') || 'Should remote work be the default for knowledge workers?';

  const client = new FireClient({ token });

  console.log(`Question: ${question}\n`);
  console.log('Starting the triad flow (hot_1 + hot_2 run in parallel)...');
  const run = await client.runFlow({ flowSlug: 'triad', input: { prompt_package: question } });
  console.log(`  run_id=${run.runId} status=${run.status}`);

  // Execution is always queued — poll until it lands somewhere interesting.
  // FlowRun#wait mutates `run` in place, so it stays current.
  await run.wait();
  console.log(`  -> ${run.status}\n`);

  if (run.status === 'awaiting_human') {
    const gate = run.gatingStep();

    console.log(`Paused for human review: ${gate.promptForHuman}\n`);
    console.log(`Agreeable take:\n  ${gate.context.hot_1}\n`);
    console.log(`Contrarian take:\n  ${gate.context.hot_2}\n`);

    const rl = createInterface({ input: stdin, output: stdout });
    const note = await rl.question('Guidance for the synthesizer (or press Enter to let it decide): ');
    rl.close();

    await run.resume(gate.stepKey, { note });
    await run.wait();
    console.log(`\n  -> ${run.status}`);
  }

  if (run.status === 'completed') {
    console.log('\n=== Final synthesized answer ===');
    console.log(run.output.content);
    console.log(`\n(cost: $${run.totalCostUsd.toFixed(6)})`);
  } else {
    console.log(`\nRun ended in status='${run.status}': ${run.error ?? ''}`);
  }
}

main();

File: python/examples/triad_example.py
#!/usr/bin/env python3
"""Runs Fire's bundled "triad" flow: two hot agents (one agreeable, one
contrarian) answer your question in parallel, you review both, then a
cool synthesizer writes the final answer.

Usage:
    FIRE_TOKEN=fire_sk_... python3 examples/triad_example.py "Should cities ban cars downtown?"

Get a token from whoever gave you access to this SDK, or self-serve one
with a tier code (see README.md's "Getting a token" section).
"""

from __future__ import annotations

import os
import sys

sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))

from fire_sdk import FireClient  # noqa: E402


def main() -> None:
    token = os.environ.get("FIRE_TOKEN")
    if not token:
        raise SystemExit("Set FIRE_TOKEN in your environment first.")

    question = " ".join(sys.argv[1:]) or "Should remote work be the default for knowledge workers?"

    client = FireClient(token=token)

    print(f"Question: {question}\n")
    print("Starting the triad flow (hot_1 + hot_2 run in parallel)...")
    run = client.run_flow(flow_slug="triad", input={"prompt_package": question})
    print(f"  run_id={run.run_id} status={run.status}")

    # Execution is always queued — poll until it lands somewhere interesting.
    # FlowRun.wait() mutates the run object in place, so `run` stays current.
    run.wait()
    print(f"  -> {run.status}\n")

    if run.status == "awaiting_human":
        gate = run.gating_step()

        print(f"Paused for human review: {gate.prompt_for_human}\n")
        print("Agreeable take:")
        print(f"  {gate.context['hot_1']}\n")
        print("Contrarian take:")
        print(f"  {gate.context['hot_2']}\n")

        note = input("Guidance for the synthesizer (or press Enter to let it decide): ")

        run.resume(gate.step_key, {"note": note})
        run.wait()
        print(f"\n  -> {run.status}")

    if run.status == "completed":
        print("\n=== Final synthesized answer ===")
        print(run.output["content"])
        print(f"\n(cost: ${run.total_cost_usd:.6f})")
    else:
        print(f"\nRun ended in status={run.status!r}: {run.error}")


if __name__ == "__main__":
    main()
